cpp/include/cudf/utilities/default_stream.hpp

```diff
@@ -16,8 +16,22 @@
 
 #pragma once
 
+#include <rmm/cuda_stream_view.hpp>
+
 namespace cudf {
 
+/**
+ * @brief Default stream for cudf
+ *
+ * Use this value to ensure the correct stream is used when compiled with per
+ * thread default stream.
+ */
+#if defined(CUDF_USE_PER_THREAD_DEFAULT_STREAM)
+static const rmm::cuda_stream_view default_stream_value{rmm::cuda_stream_per_thread};
+#else
+static constexpr rmm::cuda_stream_view default_stream_value{};
+#endif
+
```

Comment:
This code is redundant and should not be added to libcudf.  When a binary (e.g. libcudf) is compiled with CUDA_API_PER_THREAD_DEFAULT_STREAM, then stream zero is interpreted as the per-thread default stream, rather than the legacy default stream (null stream). `rmm::cuda_stream_default` is defined to be zero, and therefore will be interpreted as the per-thread default stream when compiled appropriately.

Recommended reading: https://developer.nvidia.com/blog/gpu-pro-tip-cuda-7-streams-simplify-concurrency/